yt_dlp/extractor/hockeycanada.py

```diff
@@ -0,0 +1,105 @@
+            'thumbnail': 're:^https?://wpmedia01-a.akamaihd.net/en/asset/public/image/.+',
+        },
+        'params': {'skip_download': True},
+    }, {
+        'url': 'https://video.hockeycanada.ca/en/c/mwc-remembering-the-wild-ride-in-riga.112307',
+        'info_dict': {
+            'id': '112307',
+            'title': 'MWC: Remembering the wild ride in Riga',
+            'ext': 'mp4',
+            'duration': 322,
+            'timestamp': 1716235607,
+            'upload_date': '20240520',
+            'tags': ['English', '2024', "National Men's Team", 'IIHF World Championship', 'Fan'],
+            'description': 'md5:fa853281d3e8e0b1463166dc49e975b7',
+            'thumbnail': 're:^https?://wpmedia01-a.akamaihd.net/en/asset/public/image/.+',
+        },
+        'params': {'skip_download': True},
+    }, {
+        # the same video in French
+        'url': 'https://video.hockeycanada.ca/fr/c/cmm-retour-sur-un-parcours-endiable-a-riga.112304',
+        'info_dict': {
+            'id': '112304',
+            'title': 'CMM : Retour sur un parcours endiablé à Riga',
+            'ext': 'mp4',
+            'duration': 322,
+            'timestamp': 1716235545,
+            'upload_date': '20240520',
+            'tags': ['French', '2024', "National Men's Team", 'IIHF World Championship', 'Fan'],
+            'description': 'md5:cf825222882a3dab1cd62cffcf3b4d1f',
+            'thumbnail': 're:^https?://wpmedia01-a.akamaihd.net/en/asset/public/image/.+',
+        },
+        'params': {'skip_download': True},
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        data_url = self._html_search_regex(
+            r'content_api:\s*(["\'])(?P<url>https?://.+?)\1', webpage, 'content api url', group='url')
+        media_config = traverse_obj(
+            self._download_json(data_url, video_id),
+            ('config', {base64.b64decode}, {bytes.decode}, {json.loads}, {dict}))
+
+        return {
+            'id': video_id,
+            'formats': [*self._yield_formats(media_config, video_id)],
+            **self._search_json_ld(
+                webpage.replace('/*<![CDATA[*/', '').replace('/*]]>*/', ''), video_id),
+        }
+
+    def _yield_formats(self, media_config, video_id):
+        for media_source in traverse_obj(media_config, ('media', 'source', lambda _, v: url_or_none(v['src']))):
+            media_url = media_source['src']
+            media_type = mimetype2ext(media_source.get('type'))
+
+            if media_type == 'm3u8':
+                yield from self._extract_m3u8_formats(media_url, video_id, fatal=False, m3u8_id='hls')
+            elif media_type == 'mp4':
+                fmt = {
+                    'format_id': 'http',
+                    'url': media_url,
+                    'ext': 'mp4',
+                    'vcodec': 'avc1',
+                    'acodec': 'mp4a.40.2',
```

Comment:
I don't think we should hardcode this. While it might hold true now, they could always add new MP4 formats using different codecs. If we don't know we just don't know.
```suggestion
```